torchgeo/datasets/spacenet.py

```diff
@@ -271,6 +289,55 @@ def _download(self, collections: List[str], api_key: Optional[str] = None) -> No
             print("Extracting...")
             extract_archive(archive_path)
 
+    def plot(
+        self,
+        sample: Dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+
+        .. versionadded:: 0.2
+        """
+        # image can be 1 channel or >3 channels
+        if sample["image"].shape[0] == 1:
+            image = np.rollaxis(sample["image"].numpy(), 0, 3)
+        else:
+            image = np.rollaxis(sample["image"][:3].numpy(), 0, 3)
+        image = contrast_stretch(image)
+
+        ncols = 1
+        show_mask = False
+        if "mask" in sample:
+            mask = sample["mask"].numpy()
+            ncols += 1
+            show_mask = True
+        fig, axs = plt.subplots(ncols=ncols, figsize=(ncols * 8, 8))
+        if not isinstance(axs, np.ndarray):
+            axs = [axs]
+        axs[0].imshow(image)
+        axs[0].axis("off")
+        if show_titles:
+            axs[0].set_title("Image")
+
+        if show_mask:
+            axs[1].imshow(mask)
```

Comment:
does this need a `vmin` and `vmax` to behave correctly in the case where not all classes are present in an image?

Reply:
you could probably add `interpolation="none"` so you don't get this effect:

![image](https://user-images.githubusercontent.com/1088078/142680271-ec5b04c6-39a6-426a-89b7-f3ef479dd5e1.png)

Reply:
> you could probably add `interpolation="none"` so you don't get this effect:
> 
> ![image](https://user-images.githubusercontent.com/1088078/142680271-ec5b04c6-39a6-426a-89b7-f3ef479dd5e1.png)

Setting `axs[1].imshow(mask, interpolation=None)` doesn't seem to have an effect on the plot

Reply:
> does this need a `vmin` and `vmax` to behave correctly in the case where not all classes are present in an image?

All the SpaceNet datasets (except SpaceNet5 IIRC) have one class (buildings or roads).